src/main/java/ch/njol/skript/expressions/ExprIndicesOf.java

```diff
@@ -0,0 +1,214 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * @author Peter Güttinger
+ */
+@Name("Index Of")
+@Description({
+	"The first, last or all indices of a character (or text) in a text, or -1 if it doesn't occur in the text."
+		+ "Indices range from 1 to the <a href='#ExprIndicesOf'>length</a> of the text.",
+	"You can also get the indices or positions of a list where the value at that index is the provided value. "
+		+ "Indices are only supported for variable lists and will return the string indices of the given value. "
+		+ "Positions can be used with any list and will return the numerical position of the value in the list, counting up from 1."
+})
+@Examples({
+	"set {_first} to the first index of \"@\" in the text argument",
+	"if {_s} contains \"abc\":",
+		"\tset {_s} to the first (index of \"abc\" in {_s} + 3) characters of {_s} # removes everything after the first \"abc\" from {_s}",
+	"",
+	"set {_list::*} to 1, 2, 3, 1, 2, 3",
+	"set {_indices::*} to all indices of the value 1 in {_list::*}",
+	"# {_indices::*} is now \"1\" and \"4\"",
+	"",
+	"set {_indices::*} to all indices of the value 2 in {_list::*}",
+	"# {_indices::*} is now \"2\" and \"5\"",
+	"",
+	"set {_positions::*} to all positions of the value 3 in {_list::*}",
+	"# {_positions::*} is now 3 and 6",
+	"",
+	"set {_otherlist::bar} to 100",
+	"set {_otherlist::hello} to \"hi\"",
+	"set {_otherlist::burb} to 100",
+	"set {_otherlist::tud} to \"hi\"",
+	"set {_otherlist::foo} to 100",
+	"",
+	"set {_indices::*} to the first index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"bar\"",
+	"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"foo\"",
+	"",
+	"set {_positions::*} to all positions of the value 100 in {_otherlist::*}",
+	"# {_positions::*} is now 1, 3 and 5",
+	"set {_positions::*} to all positions of the value \"hi\" in {_otherlist::*}",
+	"# {_positions::*} is now 2 and 4"
+})
+@Since("2.1, INSERT VERSION (indices, positions of list)")
+public class ExprIndicesOf extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOf.class, Object.class, ExpressionType.COMBINED,
+			"[the] [first|1:last|2:all] (indices|index[es]) of [[the] value] %object% in %~objects%",
+			"[the] [first|1:last|2:all] position[s] of [[the] value] %object% in %~objects%",
+			"[the] [first|1:last|2:all] (position[s]|indices|index[es]) of [[the] value] %string% in %string%"
```

Comment:
won't this be overshadowed by the first 2 patterns
it probably needs to be first